components/SkilledImpact/elements/CampaignContent.js

```diff
@@ -0,0 +1,94 @@
+    super(props);
+    this.state = {
+      skillExpand: true,
+      isReachable: true,
+      skills: props.skills,
+      campaigns: props.campaigns
+    };
+  }
+
+  skillExpand = () =>{
+    this.setState({skillExpand: !this.state.skillExpand})
+  };
+
+  render() {
+    const campaignList = this.state.campaigns;
+    return (
+      <View style={styles.itemContainers}>
+        <TouchableOpacity style={styles.itemTitleRow} onPress={this.skillExpand}>
+          <Lightening/>
+          <Text style={styles.itemTitleText}>
+            Current Campaigns
+          </Text>
+          <View style={styles.chevronArrowContainer}>
+            {this.state.skillExpand ? <ChevronBottom/>:<ChevronRight/>}
+          </View>
+        </TouchableOpacity>
+        {this.state.skillExpand ? (
+          <View style={styles.itemContentBody}>
+              {campaignList.map((campaign, keyIndex)=>{
+                if(campaign){
+                  return(
+                    <View key={keyIndex} style={styles.itemContentRows}>
+                    <View style={styles.avatarImageContainer}>
+                      <Avatar
+                        size={65}
+                        rounded
+                        source={{
+                          uri: campaign.image || undefined
+                        }}
+                      />
+                    </View>
+                      <View style={styles.campaignRightContainer}>
+                        <View style={styles.campaignRow}>
+                          {campaign.name? (
+                            <Text style={styles.campaignOrganizationName}>
+                              {campaign.name}
+                            </Text>
+                            ):(
+                            <Text style={styles.campaignOrganizationName}>
+                              {campaign.org_name}
+                            </Text>
+                          )}
+                          <View style={styles.chevronArrowContainer}>
+                          <ChevronRight />
+                          </View>
+                        </View>
+                        <View style={styles.campaignRow}>
+                          {campaign.is_deactivated ?
+                            (<View style={styles.closeTag}>
+                              <Text style={styles.closeText}>CLOSE</Text>
+                            </View>):
+                            (<View style={styles.openTag}>
+                              <Text style={styles.mediumButtonText}>OPEN</Text>
+                            </View>)}
+                        </View>
+                        <View style={styles.campaignRowFooter}>
+                          <Text style={styles.intervalPostedText}>{moment.utc(campaign.created_at).local().startOf('seconds').fromNow()}</Text>
+                        </View>
+                      </View>
+                    </View>
+                  );
+                }
+              })}
```

Comment:
**suggestion**: Let's build a separate component for an individual campaign entry, then map the list of campaigns to a list of those components. 45 lines is pretty long to inline all of this.

Reply:
(you can put that component in this file since we only need it here, but it should make things a lot easier to read)